talisker/metrics.py

```diff
@@ -40,11 +40,25 @@
 
 from talisker import prometheus_lock
 import talisker.statsd
+from talisker.util import pkg_version, TaliskerVersionException
 
 
 try:
     import prometheus_client
-    from prometheus_client.multiprocess import mark_process_dead
+    from prometheus_client.multiprocess import (
+        MultiProcessCollector,
+        mark_process_dead,
+    )
+    prom_version = pkg_version('prometheus_client')
+    if prom_version in ('0.4.0', '0.4.1'):
+        raise TaliskerVersionException(
+            'prometheus_client {} has a critical bug in multiprocess mode, '
+            'and is not supported in Talisker. '
+            'https://github.com/prometheus/client_python/issues/322'.format(
+                prom_version,
+            )
```

Comment:
the `.format` doesn't work like that with the magic string concatenation, you'll either have to wrap the template in ()s, or move the `.format()` on the line with the `{}` but then you don't get the magic concatenation

Reply:
Narrator: Adam was wrong.

;P